### PR TITLE
UIFixes: UI Fixes/Improvements

### PR DESCRIPTION
**Pull request by NickPhura**
## Links to Jira Tickets

None

## Description of Changes

Fixing sort menu issues

### CODE REVIEW

**Review by KjartanE — DISMISSED** [review]
😸 tested locally!

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx [hunk 1]
```diff
@@ -1,19 +1,57 @@
...
+   * @example: { name: 'Name', publish_timestamp: 'Publish Date' }
+   * @memberof ISubmissionsListSortMenuProps
+   */
   sortMenuItems: Partial<Record<keyof SubmissionRecord, string>>;
+
+  /**
+   * Manually syncs the default sort with what api responds with
+   *
+   * Note: This does not sort the submissions, only sets the default selected sort of the menu.
+   * Open to suggestions on a better way to implement this
```
Comment (curtisupshall) [inline app/src/features/submissions/list/SubmissionsListSortMenu.tsx:48]: Shouldn't the Submission List maintain the sorting order, and the Sort Menu simply dictate the order it should sort them in?

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx [hunk 2]
```diff
@@ -1,19 +1,57 @@
...
+   * The object values will be the labels for the menu items
+   *
+   * @type {Partial<Record<keyof SubmissionRecord, string>>}
+   * @example: { name: 'Name', publish_timestamp: 'Publish Date' }
+   * @memberof ISubmissionsListSortMenuProps
+   */
   sortMenuItems: Partial<Record<keyof SubmissionRecord, string>>;
+
+  /**
+   * Manually syncs the default sort with what api responds with
```
Comment (curtisupshall) [inline app/src/features/submissions/list/SubmissionsListSortMenu.tsx:45]: What does it mean to "sync the default sort with what the API responds with"?

**Review by al-rosenthal — DISMISSED** [review]
🏷️

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx (hunk 2 above)
Comment (MacDeluca) [reply app/src/features/submissions/list/SubmissionsListSortMenu.tsx:45]: Yea this was the fix for having the default selected sort be the same as the api response.

ie: The api responds with the submissions sorted differently based on each endpoint (so couldnt just default to submission_timestamp etc). This just manually sets the default selected sort for the menu. This prevents a dead click if the first menu item selected is the same as the initial sort.

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx (hunk 1 above)
Comment (MacDeluca) [reply app/src/features/submissions/list/SubmissionsListSortMenu.tsx:48]: Currently the Sort Menu handles the sorting order and the sorting of the submissions. The parent component (Submissions List) handle the state for the submissions.

I did it this way to make the Sort Menu reusable accross all the submission views.

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx (hunk 2 above)
Comment (MacDeluca) [reply app/src/features/submissions/list/SubmissionsListSortMenu.tsx:45]: A better prop name might be 'defaultSelectedSort'

Comment (curtisupshall) [reply app/src/features/submissions/list/SubmissionsListSortMenu.tsx:45]: Or maybe `initialSelectedSort`

**Review by MacDeluca — APPROVED** [review]
LOVE IT

**Review by KjartanE — APPROVED** [review]
🐈

File: app/src/features/submissions/list/SubmissionsListSortMenu.tsx (hunk 2 above)
Comment (jeznorth) [reply app/src/features/submissions/list/SubmissionsListSortMenu.tsx:45]: We can address this later as tech debt.

**Pull request merged**